controllers/update.go

```diff
@@ -0,0 +1,147 @@
+			needsUpdating = true
+		}
+	}
+	if needsUpdating {
+		log.Info("Updating the DevfileRegistry deployment")
+		return r.Update(ctx, dep)
+	}
+	return nil
+}
+
+// updateRoute checks to see if any of the fields in an existing devfile index route needs updating
+func (r *DevfileRegistryReconciler) updateRoute(ctx context.Context, cr *registryv1alpha1.DevfileRegistry, route *routev1.Route) error {
+	needsUpdating := false
+
+	// Check to see if TLS fields were updated
+	if registry.IsTLSEnabled(cr) {
+		if route.Spec.TLS == nil {
+			route.Spec.TLS = &routev1.TLSConfig{Termination: routev1.TLSTerminationEdge}
+			needsUpdating = true
+		}
+	} else {
+		if route.Spec.TLS != nil {
+			route.Spec.TLS = nil
+			needsUpdating = true
+		}
+	}
+
+	if needsUpdating {
+		return r.Update(ctx, route)
+	}
+	return nil
+}
+
+// updateIngress checks to see if any of the fields in an existing ingress resouorce need to be updated
+func (r *DevfileRegistryReconciler) updateIngress(ctx context.Context, cr *registryv1alpha1.DevfileRegistry, hostname string, ingress *v1beta1.Ingress) error {
+	needsUpdating := false
+	// Check to see if TLS fields were updated
+	if registry.IsTLSEnabled(cr) {
+		if len(ingress.Spec.TLS) == 0 {
+			// TLS was toggled on, so enable it in the ingress spec
+			ingress.Spec.TLS = []v1beta1.IngressTLS{
+				{
+					Hosts:      []string{hostname},
+					SecretName: cr.Spec.TLS.SecretName,
+				},
+			}
+			needsUpdating = true
+		}
+		if ingress.Spec.TLS[0].SecretName != cr.Spec.TLS.SecretName {
+			// TLS secret name was updated, so update it in the ingress spec
+			ingress.Spec.TLS[0].SecretName = cr.Spec.TLS.SecretName
+			needsUpdating = true
+		}
+	} else {
+		if len(ingress.Spec.TLS) > 0 {
+			// TLS was disabled, so disable it in the ingress spec
+			ingress.Spec.TLS = []v1beta1.IngressTLS{}
+			needsUpdating = true
+		}
+	}
+
+	// Check to see if the ingress domain was updated
+	if ingress.Spec.Rules[0].Host != hostname {
+		ingress.Spec.Rules[0].Host = hostname
+
+		// If TLS is enabled, need to update the hostname there too
+		if registry.IsTLSEnabled(cr) {
+			ingress.Spec.TLS[0].Hosts = []string{hostname}
+		}
+		needsUpdating = true
+	}
+
+	if needsUpdating {
+		return r.Update(ctx, ingress, &client.UpdateOptions{})
+	}
+
+	return nil
+}
+
+// deletePVCIfNeeded deletes the PVC for the devfile registry if one exists and if persistent storage was disabled
+func (r *DevfileRegistryReconciler) deletePVCIfNeeded(ctx context.Context, cr *registryv1alpha1.DevfileRegistry) error {
+	// Check to see if a PVC exists, if so, need to clean it up because storage was disabled
+	pvc := &corev1.PersistentVolumeClaim{}
+	err := r.Get(ctx, types.NamespacedName{Name: registry.PVCName(cr.Name), Namespace: cr.Namespace}, pvc)
+	if err != nil && !errors.IsNotFound(err) {
+		log.Error(err, "Error listing PersistentVolumeClaims")
+		return err
+	} else if pvc != nil && pvc.Name != "" {
```

Comment:
Why does an existing pvc with a non-empty name considered as disabled?

Reply:
So I think this function is a little confusing. It's called when `storage.enabled` is false and we use it to check if there's a pre-existing PVC that now needs to be deleted since storage has been disabled (say the user toggled `storage.enabled` from true to false and updated their custom resource). We check the pvc name as a way to see if the PVC exists (if it doesn't, the name will be empty and there's nothing for us to do). 

I think there's a couple things that can be done to improve the clarity of this function:
1) Rename to `deleteOldPVCIfNeeded`
2) Move the check for `storage.enabled == false` into this function, rather than outside it.
3) Reformat the pvc check to be like:
```
if err != nil && errors.IsNotFound(err)
   // PVC not found, so just return nil. Nothing to do
else if err != nil
    // Log an error and return, as there was some error getting the list of PVCs from the cluster
else
  // PVC found so clean it up since storage has been disabled.
```

Reply:
Updated so that the logic was cleaner